examples/v1alpha2/simple-gateway/gateway.yaml

```diff
@@ -0,0 +1,14 @@
+kind: Gateway
+apiVersion: gateway.networking.k8s.io/v1alpha2
```

Comment:
Nit: https://github.com/kubernetes-sigs/gateway-api/issues/810 recommends switching the order of these in our examples (`apiVersion` then `kind`)

Reply:
e0352b72c56e051746f20c101b7435f07ad3fc2e